lib/ansible/plugins/inventory/azure_rm.py

```diff
@@ -229,11 +230,53 @@ def parse(self, inventory, loader, path, cache=True):
 
         self._filters = self.get_option('exclude_host_filters') + self.get_option('default_host_filters')
 
-        try:
-            self._credential_setup()
-            self._get_hosts()
-        except Exception as ex:
-            raise
+        # Cache logic
+        if cache:
```

Comment:
The `cache` var here corresponds to whether or not a `- meta: refresh_inventory` task is occurring. If `self.get_option('cache')` is True when inventory is being refreshed, the cache shouldn't be read but it should be updated.

Reply:
@s-hertel I'm afraid I completely missed that case. So I'm just using the cache options from the "dynamic inventory" config.
I didn't had that use-case where you had to refresh the eventory mid-playbook (that's as far as I understand at least one of the use cases you are pointing)

Out of curiosity, IIRC this was based on GCP, does it has the same issue?

Reply:
This didn't had any traction so not sure if this is still usefull or not...

Reply:
@s-hertel I'm afraid I completely missed that case. So I'm just using the cache options from the "dynamic inventory" config.
I didn't had that use-case where you had to refresh the eventory mid-playbook (that's as far as I understand at least one of the use cases you are pointing)

Out of curiosity, IIRC this was based on GCP, does it has the same issue?

Reply:
This didn't had any traction so not sure if this is still useful or not...

Reply:
I'm probably not best to assess the usefulness since I've only dabbled in using the azure inventory plugin. Pinging the #ansible-azure freenode IRC channel might help get some traction. I was just looking through the backlog of inventory plugin PRs and took a look since it seemed straightforward.

Yes, I was referring to refresh an inventory mid-playbook. I just took a look at the GCP inventory plugin and it looks like it has the same issue. There's some documentation about the expected flow of caching here https://docs.ansible.com/ansible/latest/dev_guide/developing_inventory.html#inventory-cache. There's an example of a possible parse method in that section. The foreman and aws_ec2 inventory plugins are a couple good examples of caching.

Reply:
Just updated the code, did a simple test and it seems to be working. Although I'm not currently working with Azure so it may needs more testing